src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -1053,18 +1046,15 @@ static void footpath_remove_tile_element(TileElement* tileElement)
         }
     }
 
+    gFootpathConstructFromPosition.z = tileElement->base_height * 8;
     // Remove path
-    footpath_remove(
-        gFootpathConstructFromPosition.x, gFootpathConstructFromPosition.y, tileElement->base_height, GAME_COMMAND_FLAG_APPLY);
+    footpath_remove(gFootpathConstructFromPosition, GAME_COMMAND_FLAG_APPLY);
 
     // Move selection
-    edge = direction_reverse(edge);
-    x = gFootpathConstructFromPosition.x - CoordsDirectionDelta[edge].x;
-    y = gFootpathConstructFromPosition.y - CoordsDirectionDelta[edge].y;
-    gFootpathConstructFromPosition.x = x;
-    gFootpathConstructFromPosition.y = y;
-    gFootpathConstructFromPosition.z = z << 3;
-    gFootpathConstructDirection = edge;
+    gFootpathConstructFromPosition.x -= CoordsDirectionDelta[edge].x;
+    gFootpathConstructFromPosition.y -= CoordsDirectionDelta[edge].y;
+    gFootpathConstructFromPosition.z = z * 8;
+    gFootpathConstructDirection = direction_reverse(edge);
```

Comment:
Because edge stoppped being updated before the `{ x ,y }` assignment, this caused #10426